### ISSUE TITLE
Frame details washed off in imaged based editing

### ISSUE THREAD

**Issue by jefflao** [AlonzoLeeeooo/StableV2V#20]
Thanks for the interesting work and more importantly open sourcing it.

I was trying out the image based editing on a cloud environment with Jupyterlab. The video I use as benchmarking is 6 sec long at 30 fps and 1280x720. In order to fit into a 80GB A100 I tried both scaling and cropping the frames into 640x360, and only process the first 5-10 frames.

My goal is to substitute a specific advertisement on a sport field. I use an England flag here just to visualize any offset or distortion.  After some tweaking on the runners I was able to run the code as a Jupyter notebook on non-square inputs, but the result doesn't look quite right.

1. The details is washed off after the first two frames. Originally I prompted the paint-by-example for the first frame and the flag was completely distorted.  Following your comment in #16, I provided the edited first frame by substituting the target with openCV. This only solves the issue on the first frame or two.

First frame:
![Image](https://github.com/user-attachments/assets/42ae9598-d3eb-4933-88eb-6e91fde23623)

Resulting first frame:
![Image](https://github.com/user-attachments/assets/56be280d-3a4e-4a04-a19a-12119f077591)

Original 4th frame:
![Image](https://github.com/user-attachments/assets/76a9f2e4-3f3b-4bba-ae07-089126eb7295)

Resulting 4th frame:
![Image](https://github.com/user-attachments/assets/3e5785d2-8434-415c-adc1-76c6c0e715b0)

2. The warped editing masks aren't tracking the target object. I have been working with the test video for a while and have the editing masks segmented with SAM 2 from Meta. I tried text prompts like "A logo displayed on the white billboard" but the warped masks still couldn't track the target.

Reference mask for 4th frame:
![Image](https://github.com/user-attachments/assets/9c598a9f-408d-4037-9a6d-de890b6e704d)

The warped editing mask:
![Image](https://github.com/user-attachments/assets/e20e8e0b-0624-4e17-b4aa-2cf2d5d9a1c2)

3. The color seems to be "leaking" in the output of i2vgen-xl. In bear walking example the result is a bit red-ish, and in my situation the frames are green. I believe this is an issue on i2vgen-xl, just wonder if you know any way to improve the situation.

Another minor issue is that, i2vgenxl seems to be downloading something in the first run. Whenever I restart my VM, the first pass of step 7 in `app.py` can take 45 mins to almost two hours, despite I have cloned everything on the huggingface repo to a cloud storage. After the first run, this steps speeds up significantly to minutes. I wonder if any cache or files that can be saved to speed up the process.

Thank you in advance.

Jeff

**Comment by AlonzoLeeeooo** [comment]
Hi @jefflao ,

Thanks for your interests and kind words of our work! Here are more information, and I hope they could address your problems:
1. **For your case of replacing the flag**, I understand that this is an important application in real-world editing. Nevertheless, StableV2V (or basically most of the released video editing studies) is only good at editing videos with significant objects, where you can notice that most papers show cases with single objects and simple backgrounds. It has been a non-trivial task to handle such editing cases already. The case of replacing the flag (scene-based editing) could be quite difficult for StableV2V to handle.
2. **The washed-off details** are probably caused by poorly-simulated depth map in this case, since the motion in a view of football game is pretty complicated (for neural networks), so it fails to generate anything reasonable in such case. Such complicated motions also make our ISA process tough to track the masks.
3. **The color problem** is inherently caused by Ctrl-Adapter, which tends to cause color discrepancy due to limited training data. We have discussed such problem in Section E of [our arXiv paper](https://arxiv.org/pdf/2411.11045). I2VGen-XL seems to produce videos with consistent color. **Potential solutions** for this might be replacing the Ctrl-Adapter and I2VGen-XL with more advanced conditional frameworks, which we are currently work on for further improvement.
4. **For the slow initiating process of the Gradio demo**, it is normal since there are many model weights that need to be firstly downloaded, otherwise the pipeline won't run properly. Once they are ready, it would much faster. It seems like this is a must-to-go process and I guess all you need is waiting :)

If you meet any other problems, feel free to leave a message under this issue again! Wish you have a nice day!

Best regards

**Comment by jefflao** [comment]
Hi @AlonzoLeeeooo, thank you for the information and sorry for the late response. 

Following your advice, I change to a simpler test video with a few stable items on the table, and the result improves significantly. The warped masks and depth maps can now track the objects in the frames properly, however the background is still suffering from the washed-off issue:

Edited frame 1:
![Image](https://github.com/user-attachments/assets/792a5098-a95b-457b-b8e0-1a8da37ac919)

Edited frame 4:
![Image](https://github.com/user-attachments/assets/a98d78a0-996e-49ce-a382-d1a74b341170)

final_depth_maps for frame 4:
![Image](https://github.com/user-attachments/assets/dbca86a0-d5c3-4f8e-8fcc-6b21f459c07f)

I wonder if this is also an issue of I2VGen-XL? If that is the case, have you come by any similar model for the task?

Cheers!

**Comment by AlonzoLeeeooo** [comment]
Hi @jefflao ,

From the visualization of the final depth map, it seems that everything we proposed in StableV2V work normally. I think what you are suspecting may make sense.
For similar model to I2VGen-XL, any image-to-video generation model will theoretically be alternatives to it, such as Stable Video Diffusion (SVD), VideoCrafter. According to our experience, SVD might produce still objects in some cases. And we haven't tried VideoCrafter on this task yet. Hope these information can address your problems.

Best

**Comment by jefflao** [comment]
Thank you for the details. I will look into VideoCrafter and SVD for improvements.

This clarifies the issues I have so far, I'll close this issue. 

Cheers!

**Comment by AlonzoLeeeooo** [comment]
Great! If you have any further questions in the future, please do not hesitate to contact us again!
